modules/exploits/windows/http/desktopcentral_deserialization.rb

```diff
@@ -83,26 +82,29 @@ def check
     )
 
     unless res
-      return CheckCode::Unknown('Target is not responding to check')
+      return CheckCode::Unknown('Target did not respond to check request.')
     end
 
     unless res.code == 200 && res.body.include?('ManageEngine Desktop Central')
-      return CheckCode::Unknown('Target is not running Desktop Central')
+      return CheckCode::Unknown('Target is not running Desktop Central.')
     end
 
-    version = res.get_html_document.at('//input[@id = "buildNum"]/@value')&.text
+    build = res.get_html_document.at('//input[@id = "buildNum"]/@value')&.text
 
-    unless version
-      return CheckCode::Detected('Could not detect Desktop Central version')
+    unless (build = Integer(build) rescue nil)
```

Comment:
Good job implementing better checks here and better output lines, appreciated :)